lib/templates/src/App.vue

```diff
@@ -1,28 +1,123 @@
 <template>
   <Experiment id="app" :trials="{ audio, options }">
-    <div>Mouse tracking experiment</div>
+    <template #title>
+      <div>Mouse tracking experiment</div>
+    </template>
 
-    <template #0="{ nextScreen }">
+    <template #screens>
       <Screen :title="'Welcome'">
-        <template #0>
-          Hello!
-          <button @click="nextScreen">Start</button>
-        </template>
+        Hello!
+        <button @click="$exp.nextScreen">Start</button>
       </Screen>
-    </template>
 
-    <template #[screen]="{ trial }" v-for="screen in 2">
-      <AudioDiscriminationWithPriming
-        :priming-audio="trial.audio.primingAudio"
-        :trial-audio="trial.audio.trialAudio"
-        :option1="trial.options.option1"
-        :option2="trial.options.option2"
-        :key="screen"
-      >
-      </AudioDiscriminationWithPriming>
-    </template>
+      <Screen>
+        <ForcedChoiceInput
+          question="How did you like this?"
+          :options="[$exp.trial.options.option1, $exp.trial.options.option2]"
+          :answer.sync="answer"
+          @change:answer="
+            $exp.addResult({ answer, options: $exp.trial.options });
+            $exp.nextScreen();
+          "
+        />
+      </Screen>
+
+      <Screen>
+        <ImageSelectionInput
+          question="How did you like this?"
+          :options="[
+            { label: $exp.trial.options.option1, src: 'img/1.jpg' },
+            { label: $exp.trial.options.option2, src: 'img/2.jpg' }
+          ]"
+          :answer.sync="answer"
+          @change:answer="
+            $exp.addResult({ answer, options: $exp.trial.options });
+            $exp.nextScreen();
+          "
+        />
+      </Screen>
+
+      <Screen>
+        <TextareaInput
+          question="How did you like this?"
+          :answer.sync="answer"
+          @change:answer="
+            $exp.addResult({ answer });
+            $exp.nextScreen();
+          "
+        />
+      </Screen>
+
+      <Screen>
+        <SliderInput
+          question="How did you like this?"
+          left="bad"
+          right="good"
+          :answer.sync="answer"
+        />
+        <button
+          @click="
+            $exp.addResult({ answer });
+            $exp.nextScreen();
+          "
+        >
+          Next
+        </button>
+      </Screen>
+
+      <Screen>
+        <CompletionInput
+          text="One %s fell over a %s."
+          :options="[
+            ['horse', 'penguin'],
+            ['tree', 'icicle']
+          ]"
```

Comment:
It seems that in most of the screens, the options are not supplied via the `trials` prop on `$exp`. I'm not familiar with how the original `_magpie` frontend works. Would it be desirable to unify this somehow, so that the options for all trials are provided via a central source of truth, or would more flexibility (i.e. specifying the options on the spot) be fine. I guess it would also be a question for @michael-franke

Reply:
The idea is definitely to use the `trials` prop whenever possible. I mainly hardcoded these here, because of laziness, as it's just a demo.